code/modules/surgery/organs/augments_internal.dm

```diff
@@ -137,6 +137,37 @@
 /obj/item/organ/cyberimp/brain/anti_stun/proc/reboot()
 	broken_cyber_organ = FALSE
 
+/obj/item/organ/cyberimp/brain/abductor_interface
+	name = "Abductor Interface implant"
+	desc = "This implant will identify you as an abductor when interacting with abductor technology, as well as automatically giving the expertise needed to operate it.\
+	It will also give an innate knowledge of surgical procedures."
+	implant_color = "#FD0098"
+	slot = ORGAN_SLOT_BRAIN_ABDUCTOR
+
+/obj/item/organ/cyberimp/brain/abductor_interface/Insert()
+	. = ..()
+	ADD_TRAIT(owner, TRAIT_ABDUCTOR_TRAINING, "interface")
+	ADD_TRAIT(owner, TRAIT_ABDUCTOR_SCIENTIST_TRAINING, "interface")
+
+/obj/item/organ/cyberimp/brain/abductor_interface/Remove(mob/living/carbon/M, special = FALSE)
+	. = ..()
+	REMOVE_TRAIT(owner, TRAIT_ABDUCTOR_TRAINING, "interface")
+	REMOVE_TRAIT(owner, TRAIT_ABDUCTOR_SCIENTIST_TRAINING, "interface")
+
+/obj/item/organ/cyberimp/brain/surgeon
+	name = "Surgery Helper Implant"
+	desc = "This implant grants instinctive knowledge of advanced surgery, without requiring a surgical computer nor research."
+	implant_color = "#FD0098"
+	slot = ORGAN_SLOT_BRAIN_SURGEON
+
+/obj/item/organ/cyberimp/brain/surgeon/Insert()
+	. = ..()
+	ADD_TRAIT(owner, TRAIT_SURGEON, "implant")
+
+/obj/item/organ/cyberimp/brain/surgeon/Remove(mob/living/carbon/M, special = FALSE)
+	. = ..()
+	REMOVE_TRAIT(owner, TRAIT_SURGEON, "implant")
```

Comment:
This doesn't work. Use M instead of owner because owner is null after Remove gets called. Also, please replace the magic strings with defines. We use defines for almost all traits, even if they only have a single source.

Reply:
Almost all traits which are added and removed within two very close lines of code use straight strings; i think that adding defines for cases like these will only serve to bloat the trait defines file unnecessarily.

Reply:
There's precedent for what I said. We literally have an section of around 35 unique trait sources in traits.dm. It would just be needlessly inconsistent to not do it.

Reply:
I'm the one that did those, and i kinda regret doing it due to the inconsistency. As i said, i believe it should only be for traits that span multiple files.

Reply:
Alright, but the issue of the trait being removed from owner after owner is null still remains.

Reply:
nemvar is correct here in that they should be defines regardless of how many files they're in